File: app/src/main/java/tech/destinum/machines/di/AppComponent.java
package tech.destinum.machines.di;

import javax.inject.Singleton;

import dagger.Component;
import dagger.Provides;
import tech.destinum.machines.ACTIVITIES.Graph;
import tech.destinum.machines.ACTIVITIES.LineChart;
import tech.destinum.machines.ACTIVITIES.MachineInfo;
import tech.destinum.machines.ACTIVITIES.MainActivity;

@Singleton
@Component(modules = {AppModule.class, ViewModelModule.class})
public interface AppComponent {

    void inject(MainActivity activity);

    void inject(MachineInfo activity);

    void inject(Graph activity);

    void inject(LineChart activity);

}

File: app/src/main/java/tech/destinum/machines/DI/AppComponent.kt
package tech.destinum.machines.DI

import javax.inject.Singleton

import dagger.Component
import tech.destinum.machines.Activities.Graph
import tech.destinum.machines.Activities.LineChart
import tech.destinum.machines.Activities.MachineInfo
import tech.destinum.machines.Activities.MainActivity

@Singleton
@Component(modules = arrayOf(AppModule::class, ViewModelModule::class))
interface AppComponent {

    fun inject(activity: MainActivity)

    fun inject(activity: MachineInfo)

    fun inject(activity: Graph)

    fun inject(activity: LineChart)

}
